internal/clients/telegram/client.go

```diff
@@ -159,12 +161,31 @@ TODO: should return state for the update that will be processed.
 Should also block if the state is taken already
 (Should take some args to look up the state for the conversation)
 */
-func (c *Client) takeState() ConversationStateHandler {
-	return &rootConversationState{}
+func (c *Client) takeState(u *UpdateProcessor) ConversationStateHandler {
+	handle, err := (*u).stateHandle()
+
+	if err != nil {
+		log.Printf("Error converting an update to a state handle: %s", err)
+		return &rootConversationState{}
+	}
+
+	if s, found := c.conversationStates[handle]; found {
+		log.Println("Lent state:", s)
+		return s
+	} else {
+		return &rootConversationState{}
+	}
 }
 
 // TODO: Stores the state and allows other threads to take it via takeState
```

Comment:
this comment is still valid?

Reply:
Yes. There is no mechanism to lock the state so that `takeState()` wont give it away. Ill update the todo comment.

Reply:
https://github.com/m-kuzmin/daily-reporter/pull/4/commits/11ba4a098c951b5f88bd44593bfc77efeafab40e